src/functions.php

```diff
@@ -77,7 +77,7 @@ function uri_for($uri)
 function stream_for($resource = '', array $options = [])
 {
     switch (gettype($resource)) {
-        case 'string':
+        case is_scalar($resource):
```

Comment:
@mtdowling I'm pretty busy right now. so if anyone wants to add the test is welcomed otherwise when I got some time will do the tests also.